### ISSUE TITLE
Styled components with parent selectors are not re-rendering

### ISSUE THREAD

**Issue by andormade** [styled-components/styled-components#3401]
## Environment
styled-components@5.2.0 or newer

## Steps to reproduce
- Create a styled component with [parent selector](https://sass-lang.com/documentation/style-rules/parent-selector) in it.
- Add some conditional css
- Change the props

## Expected Behavior
- The component re-renders when the props change and displays conditional styles accordingly.

## Actual Behavior
- It renders only once and then it behaves like it was memoized.

## Reproduction
[CodeSandbox link](https://codesandbox.io/s/styled-components-parent-selector-bug-v54si)
I created two components: one with parent selector and one without. If you click the button it toggles the _isBlue_ state. You can see that first time both components change styles, but after the second toggle, the one with the parent selector in it stays the same.

If you change the version of style-components in the dependency panel back to 5.1.1 the problem disappears, and will work perfectly again.

I think it was this pull-request #3236 that caused this behavior.

**Comment by haneenmahd** [comment]
Hey Andor! I Just Fixed the bug.You can select a element with Gloablly with createGloableStyle in styled-components;
You, may try it that way.
But, When i removed your html.something selector it just worked fine.
[CodeSansBox link](https://codesandbox.io/s/cfdw3?file=/src/index.js)

**Comment by andormade** [comment]
> But, When i removed your html.something selector it just worked fine.

But that's the whole point. It supposed to work with the parent selector too.

**Comment by haneenmahd** [comment]
Yeah! It Should have!

**Comment by tu4mo** [comment]
Quite odd that the style is applied to the `sc-*`-class. When I tried it with `&&` it uses the correct class.

**Comment by kitten** [comment]
What's basically happening is that `html.something &` is not expanded as `html.something INSTANCE` anymore, but as `html.something .sc-bdfBwQ` i.e. `html.something ${SelfComponent}` so to speak.

Duplicate of #3265
